src/backend/controllers/battleController.js

```diff
@@ -0,0 +1,246 @@
+        tankTwo: challengerTank,
+        winner: -1,
+        prizeMoney: personBeingChallengedUserDoc.wager
+    });
+
+    await newRecord.save((err: Error) => {
+        if (err) {
+            console.error(err.message);
+            return res
+                .status(500)
+                .json({ msg: 'Unable to save battle record to DB.' });
+        }
+        else{
+            console.log('Match successfully created!');
+            return res
+                .status(200)
+                .send(newRecord._id) // Can I do this?
+        }
+    });
+   }catch(err){
+       return res
+               .status(500)
+               .json({ msg: 'Server Error'});
+   }
+}
+
+exports.reportResults = async (req: Request, res: Response) => 
+{
+    // Checks that we have received the correct body from the frontend 
+    const errors = validationResult(req);
+
+   if (!errors.isEmpty()) {
+       // Return 400 for a bad request
+       return res
+           .status(400)
+           .json({ errors: errors.array() });
+   }
+
+   const { winner, battleId } = req.body;
+
+   try{
+    const battle = await BattleRecord.findByIdAndUpdate(battleId, { winner : winner }, {new: true}, (err: Error) => {
+        if (err) {
+            console.error(err.message);
+            return res
+             .status(500)
+             .json({ msg: 'Failed up update battleRecord'});
+        }
+    });
+ 
+    if (battle == null) {
+        console.log('battle not found');
+        return res
+             .status(404)
+             .json({ msg: 'Could not find battle'});
+    }
+ 
+    if (winner === 0) { // tie
+        // Update the ties for user stats
+        const userOne = await User.findByIdAndUpdate(battle.userOne, { $inc: { 'stats.ties' : 1 } }, {new: true});
+        const userTwo = await User.findByIdAndUpdate(battle.userTwo, { $inc: { 'stats.ties' : 1 } }, {new: true});
+
+        if (userOne == null) {
+            console.log('userOne not found');
+            return res
+                .status(404)
+                .json({ msg: 'Could not find userOne'});
+        }
+        else if (userTwo == null){
+            console.log('userTwo not found');
+            return res
+                .status(404)
+                .json({ msg: 'Could not find userTwo'});
+        }
+    }
+    else if (winner === 1) { // userOne victory
+        
+        // Update money and wins for user one
+        const userOne = await User.findByIdAndUpdate(battle.userOne, { $inc: { money : battle.prizeMoney, 'stats.wins' : 1 } }, {new: true});
+
+        if (userOne == null) {
+            console.log('userOne not found')
+            return res
+                .status(404)
+                .json({ msg: 'Could not find userOne'});
+        }
+
+        // Update money and losses for user two
+        const userTwo = await User.findByIdAndUpdate(battle.userTwo, { $inc: { money : (battle.prizeMoney * -1), 'stats.losses' : 1 } }, {new: true});
```

Comment:
you shouldn't need to subtract out prize money here, since you should lose the money when you create the record